torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -718,6 +733,132 @@ def call_function(
                 return TupleVariable([TupleVariable(items), aux])
 
 
+class FunctorchVmapHigherOrderVariable(TorchHigherOrderOperatorVariable):
+    def call_function(
+        self, tx, args: "List[VariableTracker]", kwargs: "Dict[str, VariableTracker]"
+    ) -> "VariableTracker":
+        from . import ConstantVariable, TensorVariable
+        from .builder import wrap_fx_proxy
+
+        if not torch._dynamo.config.capture_func_transforms:
+            unimplemented("torch.func.vmap capture is disabled")
+
+        checkpoint = tx.copy_graphstate()
+        graph_checkpoint = tx.output.graph
+
+        # unpack args
+        fn = args[0]
+        in_dims = args[1]
+        out_dims = args[2]
+        randomness = args[3]
+        chunk_size = args[4]
+        batch_input_args = args[5:]
+
+        if kwargs:
+            unimplemented(
+                "NYI - torch.func.vmap: kwargs arguments are currently unsupported."
+            )
+
+        if chunk_size.value is not None:
+            unimplemented(
+                "NYI - torch.func.vmap is not implemented when chunk_size is passed"
+            )
+
+        flat_args, arg_spec = torch.utils._pytree.tree_flatten(batch_input_args)
+        in_dims_v = in_dims.as_python_constant()
```

Comment:
in_dims isn't always a constant, right?

Reply:
Probably needs a check for constant, (if not constant we should raise unimplemented)

Reply:
Wouldn't they always be Python constants (int or tuple)?